aptos-move/framework/src/natives/cryptography/ristretto255_point.rs

```diff
@@ -0,0 +1,624 @@
+
+    let a = pop_scalar_from_bytes(&mut args)?;
+
+    let basepoint = RISTRETTO_BASEPOINT_TABLE;
+
+    let result = basepoint.mul(&a);
+
+    // Compute result = a * BASEPOINT and return a RistrettoPointHandle
+    let result_handle = point_data.add_point(result);
+
+    Ok(NativeResult::ok(cost, smallvec![Value::u64(result_handle)]))
+}
+
+#[allow(non_snake_case)]
+pub(crate) fn native_basepoint_double_mul(
+    gas_params: &GasParameters,
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert_eq!(ty_args.len(), 0);
+    assert_eq!(args.len(), 3);
+
+    let cost = gas_params.base_cost + gas_params.basepoint_double_mul_cost * NumArgs::one();
+
+    let point_context = context.extensions().get::<NativeRistrettoPointContext>();
+    let mut point_data = point_context.point_data.borrow_mut();
+
+    let b = pop_scalar_from_bytes(&mut args)?;
+    let A_handle = pop_ristretto_handle(&mut args)?;
+    let a = pop_scalar_from_bytes(&mut args)?;
+
+    let A_ref = point_data.get_point(&A_handle);
+
+    // Compute result = a * A + b * BASEPOINT and return a RistrettoPointHandle
+    let result = RistrettoPoint::vartime_double_scalar_mul_basepoint(&a, A_ref, &b);
+    let result_handle = point_data.add_point(result);
+
+    Ok(NativeResult::ok(cost, smallvec![Value::u64(result_handle)]))
+}
+
+pub(crate) fn native_new_point_from_sha512(
+    gas_params: &GasParameters,
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert_eq!(ty_args.len(), 0);
+    assert_eq!(args.len(), 1);
+
+    let point_context = context.extensions().get::<NativeRistrettoPointContext>();
+    let mut point_data = point_context.point_data.borrow_mut();
+
+    let bytes = pop_arg!(args, Vec<u8>);
+
+    let cost = gas_params.base_cost
+        + gas_params.point_from_64_uniform_bytes_cost * NumArgs::one()
+        + gas_params.sha512_per_hash_cost * NumArgs::one()
+        + gas_params.sha512_per_byte_cost * NumBytes::new(bytes.len() as u64);
+
+    let result_handle = point_data.add_point(RistrettoPoint::hash_from_bytes::<Sha512>(&bytes));
+
+    Ok(NativeResult::ok(cost, smallvec![Value::u64(result_handle)]))
+}
+
+pub(crate) fn native_new_point_from_64_uniform_bytes(
+    gas_params: &GasParameters,
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert_eq!(ty_args.len(), 0);
+    assert_eq!(args.len(), 1);
+
+    let point_context = context.extensions().get::<NativeRistrettoPointContext>();
+    let mut point_data = point_context.point_data.borrow_mut();
+
+    let slice = pop_64_byte_slice(&mut args)?;
+
+    let cost = gas_params.base_cost + gas_params.point_from_64_uniform_bytes_cost * NumArgs::one();
+
+    let result_handle = point_data.add_point(RistrettoPoint::from_uniform_bytes(&slice));
```

Comment:
this is a lot which I think would be easier to read like this:
```suggestion
    let cost = gas_params.base_cost + gas_params.point_from_64_uniform_bytes_cost * NumArgs::one();

    let point_context = context.extensions().get::<NativeRistrettoPointContext>();
    let mut point_data = point_context.point_data.borrow_mut();
    let slice = pop_64_byte_slice(&mut args)?;
    let result_handle = point_data.add_point(RistrettoPoint::from_uniform_bytes(&slice));
```
or
```
    let cost = gas_params.base_cost + gas_params.point_from_64_uniform_bytes_cost * NumArgs::one();

    let point_context = context.extensions().get::<NativeRistrettoPointContext>().point_data.borrow_mut();
    let slice = pop_64_byte_slice(&mut args)?;
    let result_handle = point_data.add_point(RistrettoPoint::from_uniform_bytes(&slice));
```

Reply:
Not sure I follow: what is a lot? You just want different spacing and ordering? :-/